merkledag/merkledag.go

```diff
@@ -371,27 +410,27 @@ func EnumerateChildrenAsync(ctx context.Context, ds DAGService, root *Node, set
 func fetchNodes(ctx context.Context, ds DAGService, in <-chan []key.Key, out chan<- *Node, errs chan<- error) {
 	defer close(out)
```

Comment:
I normally avoid doing that, it causes odd race conditions, and nobody will ever be waiting solely on that error channel.

The race condition goes something like:
```
select {
case nd, ok := <-nodes:
// handle the node
case err := <-errs:
// handle the error, generally exit
}
```

if i send a node, then return and the error channel gets closed, then i might miss out on handling that node depending on how well the scheduler feels like behaving. If I only close the one channel, we only have one outcome to worry about.